commons/src/main/java/io/hyscale/commons/commands/CommandExecutor.java

```diff
@@ -122,101 +120,81 @@ public static boolean executeInDir(String command, File commandOutputFile, Strin
-     * @param file
+     * @param outputFile output stream for process {@link #getOutputStream(File)}
      * @param dir
      * @param stdInput
      * @return {@link CommandResult}
-     * @throws IOException
-     * @throws InterruptedException
      * @throws HyscaleException
+     * @throws IOException
      */
-    private static CommandResult _execute(String command, File file, String dir, String stdInput)
-            throws IOException, InterruptedException, HyscaleException {
+    private static CommandResult _execute(String command, File outputFile, String dir, String stdInput)
+            throws HyscaleException, IOException {
+        
         CommandResult cmdResult = new CommandResult();
-        int exitCode = 1;
-        cmdResult.setExitCode(exitCode);
         if (StringUtils.isBlank(command)) {
             return cmdResult;
         }
-        ProcessBuilder processBuilder = new ProcessBuilder();
         if (StringUtils.isBlank(dir) || dir.equals(".")) {
             dir = SetupConfig.CURRENT_WORKING_DIR;
         }
-        logger.debug("Executing command {} in dir {}", command, dir);
-        processBuilder.command(command.split(" "));
-        processBuilder.redirectErrorStream(true);
-        if (file != null) {
-            HyscaleFilesUtil.createEmptyFile(file);
-            processBuilder.redirectOutput(file);
-        }
-        Process process = processBuilder.start();
-        handleStandardInput(process, command, stdInput);
+        CommandLine commandToExecute = CommandLine.parse(command);
 
-        boolean readOutput = false;
-        StringWriter strWriter = new StringWriter();
-        if (file == null) {
-            readOutput = copyOutput(process, strWriter);
-        }
+        Executor executor = new DefaultExecutor();
+
+        // Output handler
+        OutputStream outputStream = getOutputStream(outputFile);
+
+        // Input handler
+        InputStream inputStream = getInputStream(stdInput);
+
+        // Stream handlers
+        PumpStreamHandler streamHandler = new PumpStreamHandler(outputStream, outputStream, inputStream);
+
+        // Timeout
+        ExecuteWatchdog watchDog = new ExecuteWatchdog(DEFAULT_TIMEOUT_IN_MILLIS);
+
+        executor.setStreamHandler(streamHandler);
+        executor.setWorkingDirectory(new File(dir));
+        executor.setWatchdog(watchDog);
+        int exitCode = 1;
         try {
-            exitCode = process.waitFor();
-        } catch (
-                InterruptedException e) {
-            logger.error("Error while waiting for process to complete");
-            throw e;
+            exitCode = executor.execute(commandToExecute);
+        } catch (ExecuteException e) {
+            // timeout case
+            exitCode = e.getExitValue();
         }
-        if (readOutput) {
-            cmdResult.setCommandOutput(strWriter.toString());
+        if (outputFile == null) {
+            cmdResult.setCommandOutput(getCommandOutput(outputStream));
```

Comment:
If user is not providing an output file use a new OutputStream Impl to write data to string.